### PR TITLE
Disabling inherited dependencies

### PR DESCRIPTION
**Pull request by hannes-harnisch**
**What does this PR do?**

Implements a new Boolean config setting `inheritdependencies` that allows disabling or explicitly enabling the inheritance of dependencies from project default settings or parent projects in Visual Studio. Under Windows by default for C++ projects, this has for example the effect of linking Windows system libraries such as kernel32 and user32 if enabled. When disabled, only explicitly specified libraries will be linked.

**How does this PR change Premake's behavior?**

It doesn't. Only new behavior.

**Anything else we should know?**

This was implemented as a response to #1733.

**Did you check all the boxes?**

- [X] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [X] Add unit tests showing fix or feature works; all tests pass
- [X] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [X] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] Minimize the number of commits
- [X] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: website/docs/inheritdependencies.md
Comment (KyrietS) [inline website/docs/inheritdependencies.md:19]: Are you sure about this one?

**Review by KyrietS — COMMENTED** [review]
I needed this config some time ago. Thank you 😁

- [ ] Also, could you mention `%(AdditionalDependencies)` somewhere in your doc page? I think this is the main reason someone would like to use this switch. Let's make it easier to find 😉

File: website/docs/inheritdependencies.md
Comment (hannes-harnisch) [reply website/docs/inheritdependencies.md:19]: What makes you unsure?

Comment (hannes-harnisch) [reply website/docs/inheritdependencies.md:19]: (Yes I'm sure)

Comment (hannes-harnisch) [reply website/docs/inheritdependencies.md:21]: Also I added a mention of the `%(AdditionalDependencies)` tag.

Comment (KyrietS) [reply website/docs/inheritdependencies.md:21]: Is this really not available on earlier versions of Visual Studio?

Comment (hannes-harnisch) [reply website/docs/inheritdependencies.md:21]: Oh that's what you meant. It probably is, but I can't test it well. It seems to exist since at earliest 2010.

Comment (hannes-harnisch) [reply website/docs/inheritdependencies.md:21]: I'd be willing to change the documentation to say "at least Visual Studio 2015" if that's acceptable.

Comment (KyrietS) [reply website/docs/inheritdependencies.md:21]: It will be closer to the truth, so change it please. Thank you! 🤗

Comment (hannes-harnisch) [reply website/docs/inheritdependencies.md]: Done.

**Comment by hannes-harnisch** [comment]
@KyrietS I'm unsure why the tests failed after merging the latest main-repo commit.

**Comment by KyrietS** [comment]
I re-run all tests and it passed. There was some connection problems (as usual). You still need a code review from some other buddy. I'm only a website guardian (of some sort) 😄

**Review by KyrietS — CHANGES_REQUESTED** [review]
Oh, almost forgot. Please add `'inheritdependencies'` to the [website/sidebars.js](https://github.com/premake/premake-core/blob/master/website/sidebars.js).

**Comment by starkos** [comment]
Hey all—sorry, I was trying to go too fast and I inadvertently merged this branch before it was approved. I backed it out but the PR is still showing as merged. @hannes-harnisch original branch is still intact.

I had intended to ask @hannes-harnisch if they could please rebase their changes against the latest master branch rather than merging master into their changes, so we don't end up with a bunch of "Merge branch 'master' into…" noise commits. If you'd like me to do this for you—since I went and made more work for you seems fair—let me know.

**Comment by hannes-harnisch** [comment]
@starkos I'm not very good with rebasing, so I might just accidentally add more noise commits...

**Comment by hannes-harnisch** [comment]
@starkos I removed the merge I did from this branch. I don't know if you can still access it easily and merge again. Tell me if you want me to do something specific.

**Pull request merged**
